scripts/recursive_build.py

```diff
@@ -0,0 +1,339 @@
+import json
+import requests
+import py7zr
+import io
+import os
+import re
+import sqlite3
+import subprocess
+
+# Path to the necto_db.db file.
+dbPath = '/home/runner/.MIKROE/NECTOStudio7/databases/necto_db.db'
+
+# Path for storing artifacts.
+testPath = '/home/runner/test_results'
+
+# Path to sdk_build_automation tool.
+toolPath = '/home/runner/MikroElektronika/NECTOStudio/bin'
+
+# Global variable to trace failed tests.
+build_failed = False
+
+# Supported compilers list for each architecture.
+compiler_list = {
+    'ARM': ['gcc_arm_none_eabi', 'clang-llvm', 'mikrocarm'],
+    'RISCV': ['xpack-riscv-none-embed-gcc', 'clang-llvm-riscv'],
+    'PIC': ['mchp_xc8', 'mikrocpic'],
+    'DSPIC': ['mchp_xc16', 'mikrocdspic'],
+    'PIC32': ['mchp_xc32', 'mikrocpic32'],
+    'AVR': ['mikrocavr']
+}
+
+# Repo information.
+REPO_OWNER = 'MikroElektronika'
+REPO_NAME = 'core_packages'
+
+# Returns the list of compilers based on the given name and type.
+def get_compilers(name):
+    if any(substring in name for substring in ["ATSAM", "STM", "TM4C", "MK"]):
+        return compiler_list["ARM"], "ARM"
+    elif "GD32" in name:
+        return compiler_list["RISCV"], "RISCV"
+    elif "PIC32" in name:
+        return compiler_list["PIC32"], "PIC32"
+    elif any(substring in name for substring in ["DSPIC", "PIC24", "dsPIC"]):
+        return compiler_list["DSPIC"], "DSPIC"
+    elif any(substring in name for substring in ["PIC18", "PIC16", "PIC12", "PIC10"]):
+        return compiler_list["PIC"], "PIC"
+    elif "AT" in name and "ATSAM" not in name:
+        return compiler_list["AVR"], "AVR"
+    
+def get_all_mcus(sdk_version):
+    mcu_list = []
+    conn = sqlite3.connect(dbPath)
+    cursor = conn.cursor()
+
+    cursor.execute(f"""
+        SELECT SDKToDevice.device_uid
+        FROM SDKToDevice
+        INNER JOIN Devices ON SDKToDevice.device_uid = Devices.uid
+        WHERE SDKToDevice.sdk_uid = '{sdk_version}'
+        AND SDKToDevice.device_uid NOT LIKE '%PIM%'
+        AND SDKToDevice.device_uid NOT LIKE '%CARD%'
+        AND SDKToDevice.device_uid NOT LIKE '%SPARKFUN%'
+        AND SDKToDevice.device_uid NOT LIKE '%SIBRAIN%';
```

Comment:
Couldn't we just use an underscore (`_`) to check stuff here, as we want only MCUs and not cards etc.

Reply:
Changed these lines according to this suggestion. Now we are just searching for Devices without any underscores in the uid.